Graph tiling

misprit7 opened the pull request:
Adds tiling method based on Fuch's graph search method. Might still have some small issues with near end indices but if this becomes an issue I'll address it in a separate PR since I want to get this merged.

misprit7 committed:
Started graph approach

misprit7 committed:
Added graph construction and conversion back to index list

misprit7 committed:
Added recursive graph search

misprit7 committed:
Finished tenetative graph implementation

misprit7 committed:
Fixed graph search specific bugs

misprit7 committed:
Fixed indexing issues

misprit7 committed:
Got tiling working for big contours

misprit7 committed:
Got tiling working on large shapes

misprit7 committed:
Merged tiling with master

misprit7 committed:
Cleaned up code for PR

tiffanymatthe commented on compile_and_install.sh:
```diff
@@ -16,8 +16,8 @@ shopt -s nocasematch
-BUILDROOT="/tmp/dcma_build"
-#BUILDROOT="build"
+# BUILDROOT="/tmp/dcma_build"
+BUILDROOT="build"
```
do we want to be changing this file?

misprit7 replied:
In theory no, but building in /tmp is really dumb since it requires a complete rebuild every time you restart (which if you use a laptop is often). If Hal really cares he can change it back, but as a default I think local building is better.

tiffanymatthe commented on src/Operations/ConvertContoursToMeshes.cc:
```diff
@@ -571,7 +571,8 @@ bool ConvertContoursToMeshes(Drover &DICOM_data,
                     }
 
                 }else if( (N_upper == 1) && (N_lower == 1) ){
-                    auto new_faces = Estimate_Contour_Correspondence(pcs.upper.front(), pcs.lower.front());
```
do we only want to apply tiling for 1 to 1?

Renu-R replied:
Yeah, I think we should replace all instance of Estimate_Contour_Correspondence

misprit7 replied:
Oh yeah that was just for testing, I think when I branched that was the only place it was used.

tiffanymatthe commented on src/Simple_Meshing.cc:
```diff
@@ -34,6 +36,395 @@
+    size_t n = N_B - 1;
+
+    // We repeat over A twice, since to search the graph we need to search repeats
+    // Don't loop over all points since last point is repeat of first
```
could we change this slightly by checking that the last point is a repeat of the first? according to Hal's implementation of closed contours, a closed contour can have the last point as the first point but it is not necessary. (I'm bringing this up because this affected my 2 to 1 branching after Dominic changed the behavior of CreateCustomContours)

misprit7 replied:
Oh I didn't know it didn't have to be a repeat. The fact that they repeat causes a lot of headache when handling indexing, but sure I can change it to check.

misprit7 committed:
Addressed PR comments

misprit7 committed:
Fixed merge conflicts

misprit7 committed:
Fixed indexing issues

misprit7 commented:
I've addressed all your comments, as well as fixing the existing indexing issues. Tiffany's manifold checks now report that my meshing is manifold, tiling should be fully ready to merge now.
